src/AbstractStructArrayBase.php

```diff
@@ -251,15 +251,15 @@ private function getInternArrayOffset(): int
      * @param bool $internCall indicates that methods is calling itself
      * @return AbstractStructArrayBase
      */
-    private function initInternArray(array $array = [], bool $internCall = false): self
+    private function initInternArray($array = [], bool $internCall = false): self
```

Comment:
It's not supposed to be anything else than an array if the class inheriting from it is defined as expected. To be discusssed within the https://github.com/WsdlToPhp/PackageGenerator/issues/248 issue

Reply:
As can be seen below, the generated class (ArrayDadosFuncionais) is a child (extends) of AbstractStructArrayBase This class (ArrayDadosFuncionais) when initialized, calls the function initInternArray with no parameters, which in turn calls the same function again with $this->getPropertyValue($this->getAttributeName()) as the first parameter. This parameter is resolved (please, see the code below) in the "dadosFuncionais" which is typed as \SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayOfDadosFuncionais (other generated class).

The attribute $dadosFuncionais, typed as ArrayOfDadosFuncionais, extends AbstractStructArrayBase, which in turn implements StructArrayInterface (StructInterface, ArrayAccess, Iterator, Traversable, Countable, iterable), StructInterface, JsonSerializable. 

And when this occurs this exceptions is raised:

WsdlToPhp\PackageBase\AbstractStructArrayBase::initInternArray(): Argument #1 ($array) must be of type array, SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayOfDadosFuncionais given, called in /app/vendor/wsdltophp/packagebase/src/AbstractStructArrayBase.php on line 262



=============================================================
<?php

namespace SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ServiceType;

use SoapFault;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayDadosBancarios;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayDadosDDP;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayDadosDependentes;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayDadosFuncionais;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayDadosPA;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayDadosRepresentanteLegal;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayDadosSICAJ as ArrayDadosSICAJAlias;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayDadosUorg;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayOfArrayDadosAfastamento;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayOfArrayDadosFinanceiros;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayPensaoRecebida;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayPensoesInstituidas;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\StructType\ArrayDadosAfastamento;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\StructType\ArrayDadosFinanceiros;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\StructType\DadosCurriculo;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\StructType\DadosDocumentacao;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\StructType\DadosEnderecoResidencial;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\StructType\DadosEscolares;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\StructType\DadosPessoais;
use SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\StructType\TotalVagasOrgao;
use \WsdlToPhp\PackageBase\AbstractSoapClientBase;

/**
 * This class stands for Consulta ServiceType
 *
 * @subpackage Services
 */
class Consulta extends AbstractSoapClientBase
{


/**
     * Method to call the operation originally named consultaDadosFuncionais
     *
     * @uses   AbstractSoapClientBase::getSoapClient()
     * @uses   AbstractSoapClientBase::setResult()
     * @uses   AbstractSoapClientBase::getResult()
     * @uses   AbstractSoapClientBase::saveLastError()
     * @param  string $siglaSistema
     * @param  string $nomeSistema
     * @param  string $senha
     * @param  string $cpf
     * @param  string $codOrgao
     * @param  string $parmExistPag
     * @param  string $parmTipoVinculo
     * @return ArrayDadosFuncionais|bool
     */
    public function consultaDadosFuncionais(string $siglaSistema, string $nomeSistema, string $senha, string $cpf, string $codOrgao, string $parmExistPag, string $parmTipoVinculo): ArrayDadosFuncionais|bool
    {
        try {
            $this->setResult($this->getSoapClient()->consultaDadosFuncionais($siglaSistema, $nomeSistema, $senha, $cpf, $codOrgao, $parmExistPag, $parmTipoVinculo));
            return $this->getResult();
        } catch (SoapFault $soapFault) {
            $this->saveLastError(__METHOD__, $soapFault);
            return false;
        }
    }
=============================================================
<?php

namespace SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType;
use WsdlToPhp\PackageBase\AbstractStructArrayBase;

/**
 * This class stands for ArrayDadosFuncionais ArrayType
 *
 * @subpackage Arrays
 */
class ArrayDadosFuncionais extends AbstractStructArrayBase
{
    /**
     * The dadosFuncionais
     * Meta information extracted from the WSDL
     * - minOccurs: 0
     * - nillable: true
     *
     * @var \SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType\ArrayOfDadosFuncionais
     */
    public $dadosFuncionais;
   // .... 
    /**
     * Returns the attribute name
     *
     * @see    AbstractStructArrayBase::getAttributeName()
     * @return string dadosFuncionais
     */
    public function getAttributeName():string
    {
        return 'dadosFuncionais';
    }

=============================================================
<?php

namespace SuppCore\SigepeBackend\Integracao\DadoPessoal\Models\ArrayType;

use WsdlToPhp\PackageBase\AbstractStructArrayBase;

/**
 * This class stands for ArrayOfDadosFuncionais ArrayType
 *
 * @subpackage Arrays
 */
class ArrayOfDadosFuncionais extends AbstractStructArrayBase
{

Reply:
The code you pasted seems incomplete. Normally, the constructor, `__construct`, has one argument which is typed as an array with an array as default value:
```php
public function __construct(array $dadosFuncionais = array())
    {
        $this
            ->setDadosFuncionais($dadosFuncionais);
    }
```
This initialize the property as an array which allows to right away loop on the object using foreach.

To loop through the items using foreach (or for in twig), until PackageBase:5.0, you have to first call the `$instance->initInternArray();` method. In my case, it does not throw an exception.

Please paste the whole code using gist if necessary